app/src/main/java/io/gnosis/safe/ui/assets/coins/CoinsFragment.kt

```diff
@@ -69,14 +68,8 @@ class CoinsFragment : BaseViewBindingFragment<FragmentCoinsBinding>() {
                                 if (adapter.itemCount == 0) {
                                     binding.contentNoData.root.visible(true)
                                 }
-                                when (action.error) {
-                                    is Offline -> {
-                                        snackbar(requireView(), R.string.error_no_internet)
-                                    }
-                                    else -> {
-                                        snackbar(requireView(), action.error.getErrorResForException())
-                                    }
-                                }
+                                val error = action.error.toError()
+                                snackbar(requireView(), error.message(requireContext(), R.string.error_description_assets_coins))
```

Comment:
The error messages are now so long it is hard to read them completely even with Snackbar.LENGTH_LONG. Could we go with LENGTH_INDEFINITE?

Reply:
I would leave it like this for now and see what users will say. We can adjust it later.

Reply:
Have you tried it? Were you able to read the full message before it disappeared?
Setting this to 6000ms would make it disappear on its own. The user can swipe it to the right if he wants to get rid of it earlier.

Reply:
let's do this as a separate task https://app.zenhub.com/workspaces/safe-multisig---mobile-5c1b98250e13551e8aae81eb/issues/gnosis/safe-android/1107